crates/build-utils/src/lib.rs

```diff
@@ -0,0 +1,28 @@
+use std::{env, fs, path::Path};
+
+// Generate a Rust source file that contains the name and version of the SDK.
+pub fn gen_name_and_sdk_version(name: &str, sdk_dep_name: &str) {
+    let workspace_dir = env::var("CARGO_WORKSPACE_DIR").unwrap();
+    let lock = fs::read_to_string(Path::new(&workspace_dir).join("Cargo.lock"))
+        .expect("Cargo.lock not found");
+
+    let version = lock
+        .split("[[package]]")
+        .find(|pkg| pkg.contains(&format!("name = \"{sdk_dep_name}\"")))
```

Comment:
This LGTM, I haven't tried it, but maybe using `cargo_metadata::MetadataCommand;` from the cargo_metadata crate might be more robust?

Reply:
I'll take a look!